tests/src/test/java/docs/javadsl/AtLeastOnceTest.java

```diff
@@ -200,4 +202,44 @@ public void consumerOneProduceConditional() throws Exception {
         tuple.first().shutdown().toCompletableFuture().get(5, TimeUnit.SECONDS), is(Done.done()));
     assertThat(tuple.second().toCompletableFuture().get(5, TimeUnit.SECONDS).size(), is(10));
   }
+
+  @Test
+  public void shouldSupportBatchingOfOffsetsWithOffsetContext() throws Exception {
+    ConsumerSettings<String, String> consumerSettings =
+        consumerDefaults().withGroupId(createGroupId());
+    String topic1 = createTopic(1);
+    String topic2 = createTopic(2);
+    Consumer.DrainingControl<Done> control =
+        Consumer.sourceWithOffsetContext(consumerSettings, Subscriptions.topics(topic1))
+            .grouped(5)
+            .map(
+                records -> {
+                  String key = records.iterator().next().key();
+                  String value =
+                      records.stream().map(ConsumerRecord::value).collect(Collectors.joining(","));
+                  return ProducerMessage.single(new ProducerRecord<>(topic2, key, value));
+                })
+            .mapContext(ConsumerMessage::createCommittableOffsetBatch)
```

Comment:
In case of `def createCommittableOffsetBatch(offsets: java.util.List[Committable])` it wouldn't compile.

Reply:
:+1: